juego.py

```diff
@@ -399,15 +395,15 @@ def play(self, level):
             if not self.running:
                 break
             if float(current_time) <= 00.10:
-                game_over()
+                # game_over() # it's undefined
```

Comment:
There's a definition of `game_over` on line 464.

Reply:
The issue with the "game_over()" function in the play method of the Game class is that it is defined after it is called, which will result in a NameError when the code is executed. However, since the game_over function is defined after the if statement, Python will not know about the function at the time it is called and raise a NameError. We could move "game_over()" definition outside of the "play()" function definition.

Reply:
`game_over` is a function defined within the `play` function.  It is a forward reference.

I disagree with this change and your justification of the change.  flake8 does cite `juego.py:402:17: F821 undefined name 'game_over'`, but that would appear to be a deficiency in flake8.  We have not seen a NameError.  Have you reproduced it?

Reply:
```python
class Test():
    def play(self):
        gameover()

        def gameover():
            print("gameover")

x = Test()
x.play()
```

yields

```log
Traceback (most recent call last):
  File "test.py", line 9, in <module>
    x.play()
  File "test.py", line 3, in play
    gameover()
UnboundLocalError: local variable 'gameover' referenced before assignment
```
no sign of NameError.

```python
class Test():
    def play(self):
        def gameover():
            print("gameover")

        gameover()

x = Test()
x.play()
```

yields "gameover".

When was this regression introduced?